src/phases/evolution-phase.ts

```diff
@@ -220,7 +222,22 @@ export class EvolutionPhase extends FormChangeBasePhase {
     this.pokemonNewFormSprite.setVisible(true);
     animations.doCircleInward(this.baseBgImg, this.container);
 
-    const onEvolutionComplete = (): void => {
+    const showStarterUnlockText = (unlockedStarters: Species[]): Promise<void> => {
+      return new Promise<void>((resolve) => {
+        if (unlockedStarters.length === 0) {
+          return resolve();
+        }
+        ui.showText(
+          i18next.t("battle:addedAsAStarter", { pokemonName: getPokemonSpecies(unlockedStarters.shift()).getName() }),
+          null,
+          () => showStarterUnlockText(unlockedStarters).then(() => resolve()),
+          null,
+          true,
+        );
+      });
+    };
```

Comment:
```suggestion
    async function showStarterUnlockText(unlockedStarters: Species[]): Promise<void> {
      for (const speciesId of unlockedStarters) {
        globalScene.playSound("level_up_fanfare");
        await new Promise<void>(resolve => {
          ui.showText(
            i18next.t("battle:addedAsAStarter", { pokemonName: getPokemonSpecies(speciesId).getName() }),
            null,
            () => resolve(),
            null,
            true,
          );
        });
      }
    };
```
Less recursion needed

Reply:
this is much better, thank you